frontend/public/components/utils/file-input.tsx

```diff
@@ -21,19 +31,57 @@ export class FileInput extends React.Component<FileInputProps, FileInputState> {
     };
     reader.readAsText(file, 'UTF-8');
   }
+  onFileDrop(event) {
+    this.setState({ dropzoneActive: false });
+    this.readFile(event[0]);
```

Comment:
Can you double check there's nothing else we need to do here?

https://github.com/react-dropzone/react-dropzone#word-of-caution-when-working-with-previews

> Important: react-dropzone doesn't manage dropped files. You need to destroy the object URL yourself whenever you don't need the preview by calling window.URL.revokeObjectURL(file.preview); to avoid memory leaks.

I think we might need `disablePreview` if we're not using previews.